bluepysnap/query.py

```diff
@@ -97,23 +106,26 @@ def _positional_mask(data, ids):
     return mask
 
 
-def _circuit_mask(data, population_name, queries):
+def _circuit_mask(data, population_name, population_type, queries):
     """Handle the population, node ID queries."""
     populations = queries.pop(POPULATION_KEY, None)
+    types = queries.pop(POPULATION_TYPE_KEY, None)
     if populations is not None and population_name not in set(utils.ensure_list(populations)):
         ids = []
+    elif types is not None and population_type not in set(utils.ensure_list(types)):
```

Comment:
Building a set from the list here isn't useful, because it requires to iterate over the list.
Using `population_type not in utils.ensure_list` is enough (the same applies to the line above for populations, even it wasn't changed in this PR).

Reply:
True, copy-pasta.
-> fixed